pkg/apisix/cluster.go

```diff
@@ -301,6 +314,74 @@ func (c *cluster) HasSynced(ctx context.Context) error {
 	}
 }
 
+// syncSchema syncs schema from APISIX regularly according to the interval.
+func (c *cluster) syncSchema(ctx context.Context, interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for {
+		if err := c.syncSchemaOnce(ctx); err != nil {
+			log.Warnf("failed to sync schema: %s", err)
+		}
+
+		select {
+		case <-ticker.C:
+			continue
+		case <-ctx.Done():
+			break
```

Comment:
Not `break`, it should be `return`, `break` just let the flow jump out the `select` block but still in the `for` block.